### DISCUSSION TITLE
Troubleshooting V5 build

### DISCUSSION THREAD

**Discussion by v-flashpoint** [sanni/cartreader#432] [Need help]
Hi all,

So, I've just finished my V5 build, and while it turns on (yay!), and I can go rhough the menus, I haven't yet managed to successfully dump a game. 

For Game Boy games, it always seems to give me a checksum error. I'm guessing I've got a naff solder somewhere on the GB port.

However, for the slots on the top board, I'm getting much less. I've tried N64, SNES, Mega Drive and NES. The first three just give me a black screen as soon as I've chosen the platform from the menu, while NES lets me configure my settings, then jams up on "searching db".

I just wanted to check with the experts before I start investigating - is it likely I've made a duff solder on all of the cartridge slots? Or is there possibly more likely something on the bridge between the two layers or something I ought to look at first?

Thanks!

**Comment by Ancyker** [comment]
If you can understand the schematics and all, you can test the pins in the slot vs somewhere on the board (before any components) to check for continuity, and if there is continuity check how much resistance there is. If resistance is high then it's probably a bad solder joint somewhere.

**Reply by v-flashpoint** [reply]
Thanks for the mega speedy reply! I'm afraid I really don't know where to start with testing (this is the first thing I've made in ages). I guess I'd better get googling how to test for continuity!

**Reply by Ancyker** [reply]
You would need a multimeter to do that. You set it to continuity (usually displayed as a speaker or sideways wifi-looking symbol) and then just touch the probes to either end of the circuit (polarity/color doesn't matter for this type of testing). Regardless of the results, you can then test ohms, note that for most meters you won't have continuity if the resistance is too high so it's possible the continuity test will fail but you'll get a reading when measuring resistance. That means something isn't right, a bad connection somewhere. That could be a solder joint or a bad pin/socket for a pin. Really depends. Once you've isolated a circuit path on the board you can start to narrow it down. You can search "PCB trace repair" on YouTube to see videos of people trying to fix traces. You likely don't have a bad trace but the troubleshooting steps to find the bad trace are basically the same as what you are doing here, you are just searching out a bad connection instead of a bad trace.

**Reply by Ancyker** [reply]
Also, if all the continuity and resistance tests pass then check using continuity for things that aren't supposed to have continuity. Check neighboring pins to make sure they aren't bridged somewhere and check for shorts to ground (by testing for continuity between non-ground pins and ground).

**Comment by v-flashpoint** [comment]
Coincidentally (or perhaps not?) my status LED isn't working either. Wonder if that's the cause of the problems (although I doubt it)?

**Reply by Icesythe7** [reply]
Pictures would help (of the solder points)

**Comment by sanni** [comment]
Since all slots are wired in parallel, trouble with the GB slot will make all other slots not work either. So first you should test GB games without the six slot adapter PCB attached. If they dump correctly you know that at least the GB slot is ok. If not you need to fix the GB slot first before attaching the six slot adapter.

**Reply by v-flashpoint** [reply]
Many thanks for this. Turns out I did have some bad solders on the game boy port, and that's all working great now!

Unfortunately, nothing on the top deck seems to be working quite right, with the same result as originally reported. I'm trying to test the resistance of the solders on the bridging pins, by touching one end to the pin, and the other to the blob of solder on top, but they all seem to be OK.

Is there anything that's most likely to be causing the issues I'm experiencing? Is it the case where one slot not working on top might mean they all don't work? 

Thanks!

Ian

**Reply by sanni** [reply]
Black screen kinda sounds as if the database search is locking up. Can you try a different SD card?

**Reply by v-flashpoint** [reply]
Thanks for the mega fast reply!

Just tried a different SD, unfortunately with the same issue (black screen as soon as I chose e.g. "N64 Cart").

 Without the six slot pcb attached, if you choose N64, should it tell you there's no cartridge attached, or would that just give you a black screen?

 I'm kind of thinking I must have a bad solder on the pins that link top to bottom somewhere. Maybe something that isn't totally bad, but just isn't quite good enough?

Is there a best way to test the quality of my solders on those pins?

**Reply by sanni** [reply]
Without the six slot adapter it should say "Gamepak Error" on the screen when selecting N64 cart.
Also try with the switch set to 5V instead of 3.3V just to cross one more thing off.

**Reply by v-flashpoint** [reply]
Thanks for the help! I've actually just made a bit of progress! Now it tells me the name of the cartridge (e.g. Rocket Robot on Wheels), but gives me a gamepak error and says cartridge unknown. Tried a few N64 games with the same results. SNES seems to work fine! Wonder what might be causing that? The games dump perfectly if you choose the ROM size manually though!

**Reply by sanni** [reply]
Then you have an issue with the database txt files on the SD. 

Download the latest release again, flash it to the Arduino, copy the included txt files from the SD folder to the SD card's root. 
Use a Windows PC to do all that just in case, since the txt files use Windows style line endings and I don't know if Linux and Mac change them.

**Comment by bpm5001** [comment]
@v-flashpoint I'm having similar issues to what you were having. No status LED, N64 and SNES options going to blank screen. Do you have any info on anything you changed to get your SNES reads to work?

I've tested continuity from the pins on my slots to the main pcb and everything seems to be good in that regard. I also can get the LED to light up if I connect the ground side to another ground on the PCB. I'm not sure if the LEDs I got would require no or different resistors, but the whole circuit for the LED seems to have no power.

**Reply by v-flashpoint** [reply]
For me, it was a variety of things! 

As sanni said, getting the gba slot to work was definitely the first step. Once that was definitely working, it turned out at least part of my issue was my SD card - when you clicked read cart, the screen would go black and it'd stay that way for a long time, to the point where I thought it had crashed - but if I left it for a minute or so, eventually it'd load the database and give me the info about the cart I'm about to dump. So switching the SD card for another fixed that problem, now it just takes a second or two.

At around that time, the LED started working too. I don't think I actually changed anything to make it work!

So with the GBA port working fine, and the new SD card, everything seemed to be working pretty much perfectly. Reflashing the arduino with the latest build fixed the n64 database issues, and then everything was good to go!

**Reply by v-flashpoint** [reply]
Also potentially worth mentioning, make sure you get the txt file game info databases from https://github.com/sanni/cartreader/releases, then just drag and drop them to your sd

I think github is a bit of a git to work with if you're not used to it (which I'm not!) , so I think to begin with I'd right clicked the files in the list here https://github.com/sanni/cartreader/tree/master/sd, which it turns out was taking the full html page rather than the txt file, which was possibly why it was just going black and taking soooooo long to try and read them. So using the version from the release list is the way to do it!
